chia/wallet/wallet_state_manager.py

```diff
@@ -754,15 +754,24 @@ async def spend_clawback_coins(self, clawback_coins: Dict[Coin, ClawbackMetadata
                 if self.main_wallet.secret_key_store.secret_key_for_public_key(derivation_record.pubkey) is None:
                     await self.main_wallet.hack_populate_secret_key_for_puzzle_hash(derivation_record.puzzle_hash)
                 amount = uint64(amount + coin.amount)
+                memos: List[bytes] = [] if len(incoming_tx.memos) == 0 else incoming_tx.memos[0][1].copy()
+                if (
+                    len(memos) > 0
+                    and not is_recipient
+                    and await self.puzzle_store.puzzle_hash_exists(recipient_puzhash)
+                ):
+                    # Sender and recipient belong to the same wallet
+                    # Clean hint in this case to prevent duplicate processing
+                    memos[0] = b""
+                if len(memos) == 1 and memos[0] == b"":
+                    memos = []
```

Comment:
We can't just drop the first element here since the second one will be the hint which is the actual memo. GUI will hide the first element.

Reply:
Why would the GUI hide the first element? Did we introduce this for this clawback case now or is there some general rule we established, that we include empty bytes for the first memo if its not a hint? Im not aware of this. Maybe the GUI should be adjusted instead if this is the only case where this is relevant?

Reply:
We set the first element as b"" majorly because we don't want to mess up others' wallets. If the second element is someone's puzzle hash, the wallet will process this coin but it will throw error since it doesn't own the puzzle hash. The GUI thing exists since the first day I think. Since the first element is set by the wallet, so we don't want to show the wallet generated memo unless it is the only element.

Reply:
> We set the first element as b"" majorly because we don't want to mess up others' wallets.

Can you link me to some other place where we do this? Maybe i just always missed it but i never saw it before.

> If the second element is someone's puzzle hash, the wallet will process this coin but it will throw error since it doesn't own the puzzle hash. 

Why/How would this be the case?

Reply:
Everywhere we use hint will have this issues, for examples
For CAT: https://github.com/Chia-Network/chia-blockchain/blob/release/1.8.2/chia/wallet/wallet_state_manager.py#L862
For DID: https://github.com/Chia-Network/chia-blockchain/blob/release/1.8.2/chia/wallet/wallet_state_manager.py#L917
It doesn't mean we add coins to wrong wallets, it just generate some unnecessary error logs in these wallets.

Reply:
My point is, this is not something new and we already now on main in every other case don't know if the first entry, if its a puzzle hash, belongs to us or not so i wonder why should we now in case of clawback start to change the way we handle things usually and introduce a empty first entry? Its seems very unlikely to me that someone puts someone else's puzzle hash into the memo or do i miss something here?

Reply:
For The GUI question, you can try it by sending a CAT with a memo. Every CAT has a hint as the first element and your memo will be the second element. On GUI you will only see the second element.

Reply:
> For The GUI question, you can try it by sending a CAT or Clawback tx with a memo. Every CAT has a hint as the first element and your memo will be the second element. On GUI you will only see the second element.

Okay yeah but a CAT is supposed to have a hint in any case, a clawed/claimed XCH coin not?

Reply:
As I mentioned above, the empty byte is not for GUI majorly. Assume the memo list is [My_Puzzle_Hash, Your_Puzzle_HASH]. In the common case, your wallet will not process this coin state because the hint is my puzzle hash. But if you drop the first element, then the hint will be your puzzle hash and your wallet will try to process this coin states. But since it doesn't match the actual inner p2 puzzle so it will fail and log an error. In the worst case, it may trigger some unknown bug. So if we want to prevent this unnecessary processing, we need to set the first element to empty or a byte32 that doesn't belong to any wallet.

Reply:
Did you read this https://github.com/Chia-Network/chia-blockchain/pull/15496#discussion_r1242716607?

Reply:
I agree it's a rare case, I will change this. We'd better merge this before the release.